packages/core/src/router/routing-table.ts

```diff
@@ -216,43 +214,61 @@ export class Route extends BaseRoute {
   }
 }
 
+type ControllerInstance = {[opName: string]: Function};
+
 export class ControllerRoute extends BaseRoute {
+  protected readonly _methodName: string;
+
   constructor(
     verb: string,
     path: string,
     spec: OperationObject,
-    protected readonly _controllerName: string,
+    protected readonly _controllerCtor: ControllerClass,
+    methodName?: string,
   ) {
     super(verb, path, spec);
-  }
 
-  private get _controllerBindingKey() {
-    return `controllers.${this._controllerName}`;
+    if (!methodName) {
+      methodName = this.spec['x-operation-name'];
+    }
+
+    if (!methodName) {
```

Comment:
The current implementation throws only when both `methodName` argument and `this.spec['x-operation-name']` are empty. In your proposal, we would effectively ignore `x-operation-name` and throw whenever `methodName` is empty.